src/strategy/dungeons/wotlk/forgeofsouls/ForgeOfSoulsActions.cpp

```diff
@@ -0,0 +1,87 @@
+#include "Playerbots.h"
+#include "ForgeOfSoulsActions.h"
+#include "ForgeOfSoulsStrategy.h"
+
+bool MoveFromBronjahmAction::Execute(Event event)
+{    
+    Unit* boss = AI_VALUE2(Unit*, "find target", "bronjahm");
+    if (!boss)
+        return false;
+
+    
+    float distance = bot->GetExactDist2d(boss->GetPosition());
+    float targetDis = 20.0f;
+    if (distance >= targetDis)
+        return false;
+    return MoveAway(boss, targetDis - distance);
+}
+
+
+bool AttackCorruptedSoulFragmentAction::Execute(Event event)
+{
+    Unit* fragment = nullptr;
+
+    GuidVector targets = AI_VALUE(GuidVector, "possible targets no los");
+
+    for (auto &target : targets)
+    {
+        Unit* unit = botAI->GetUnit(target);
+        if (unit && unit->GetEntry() == NPC_CORRUPTED_SOUL_FRAGMENT)
+        {
+            fragment = unit;
+            break;
+        }
+    }
+
+    if (fragment && AI_VALUE(Unit*, "current target") != fragment)
+        return Attack(fragment);
+
+    return false;
+    
+}
+
+
+bool BronjahmGroupPositionAction::Execute(Event event)
+{
+    Unit* boss = AI_VALUE2(Unit*, "find target", "bronjahm");
+    if (!boss)
+        return false;
+
+    if (botAI->IsTank(bot))
+    {
+        bot->SetTarget(boss->GetGUID());
+        if (AI_VALUE2(bool, "has aggro", "current target"))
+            if (bot->GetExactDist2d(BRONJAHM_TANK_POSITION) > 5.0f)
+                return MoveTo(bot->GetMapId(), BRONJAHM_TANK_POSITION.GetPositionX(),
+                              BRONJAHM_TANK_POSITION.GetPositionY(), BRONJAHM_TANK_POSITION.GetPositionZ(), false,
+                              false, false, true, MovementPriority::MOVEMENT_NORMAL);
+            else
+                return Attack(boss);
+        else
+        {
+            return Attack(boss);
```

Comment:
This probably shouldn't be happening here, the tank will manage his target on his own, you do not want to be mixing attack commands in with the movement and positioning.
See IonarTankPositionAction for HoL, you should only need to move the tank and keep him there, the AI should do the rest: https://github.com/liyunfan1223/mod-playerbots/blob/7fa1ab36a3af189cb439d318fdf0fa67d849fe6e/src/strategy/dungeons/wotlk/hallsoflightning/HallsOfLightningActions.cpp#L111